src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala

```diff
@@ -2206,5 +2206,63 @@ class AppsControllerTest extends UnitTest with GroupCreation with ScalatestRoute
         header[Headers.`Marathon-Deployment-Id`] should not be 'empty
       }
     }
+    "Kill tasks and scale" in new Fixture {
+      val app = AppDefinition(id = PathId("/app"))
+      val rootGroup = createRootGroup(Map(app.id -> app))
+      val plan = DeploymentPlan(rootGroup, rootGroup)
+      groupManager.app(PathId("/app")) returns Some(app)
+
+      groupManager.rootGroup() returns rootGroup
+      taskKiller.killAndScale(any, any, any)(any) returns Future.successful(plan)
```

Comment:
Here and below I guess it could be at least `taskKiller.killAndScale(app.id, any...)` to make sure we're not trying to kill anything else. Even if that's a bit paranoid.